search_service/proxy/atlas.py

```diff
@@ -45,71 +35,187 @@ def __init__(self, *,
-            db_name = table_qn.get("db_name", '')
-            db_cluster = table_qn.get("cluster_name", '')
-
-            tags = []  # type: List[Tag]
-            # Using or in case, if the key 'classifications' is there with attrs None
-            for classification in table_attrs.get("classifications") or list():
-                tags.append(Tag(
-                    tag_name=classification.get('typeName')))
-
-            # TODO need to populate these
-            badges = []  # type: List[Tag]
-
-            # TODO: Implement columns: Not sure if we need this for the search results.
-            columns: List[str] = []
-            # for column in attrs.get('columns') or list():
-            #     col_entity = entity.referredEntities[column['guid']]
-            #     col_attrs = col_entity['attributes']
-            #     columns.append(col_attrs.get(self.NAME_KEY))
-            # table_name = attrs.get(self.NAME_ATTRIBUTE)
-            table = Table(name=table_name,
-                          key=f"{table.typeName}://{db_cluster}.{db_name}/{table_name}",
-                          description=table_attrs.get('description'),
+
+        result = list()
+
+        # if condition is satisfied then we query Atlas again to collect all available information regarding each table
+        # along with relationship information. This is helpful when using Atlas DSL as returned entities contain minimal
+        # amount of attributes.
+        if enhance_metadata:
+            ids = list()
+
+            for hit in response:
+                ids.append(hit.guid)
+
+            entities = self._extract_entities(self.atlas.entity_bulk(guid=ids, ignoreRelationships=False))
+        else:
+            entities = response
+
+        for entity in entities:
+            entity_attrs = entity.attributes
+
+            qn = parse_table_qualified_name(qualified_name=entity_attrs.get(self.ATLAS_QN_ATTRIBUTE))
+
+            entity_name = qn.get('table_name') or entity_attrs.get('name')
+            db_name = qn.get('db_name', '')
+            db_cluster = qn.get('cluster_name', '')
+
+            tags: List[Tag] = []
+
+            for classification in entity.classificationNames or list():
+                tags.append(Tag(tag_name=classification))
+
+            badges: List[Tag] = tags
+
+            table = Table(name=entity_name,
+                          key=f"{entity.typeName}://{db_cluster}.{db_name}/{entity_name}",
+                          description=entity_attrs.get('description'),
                           cluster=db_cluster,
-                          database=table.typeName,
+                          database=entity.typeName,
                           schema=db_name,
-                          column_names=columns,
                           tags=tags,
                           badges=badges,
-                          last_updated_timestamp=table_attrs.get('updateTime'))
+                          column_names=[],
```

Comment:
Is this expected? The column names being empty?

Reply:
I don't think columns are used anywhere in results view so I think it's fine.